### PR TITLE
Docs: Show code for top example and Do's

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Added the ability to see the code for our top-most header examples on component pages
Also added a way to see the code for Best Practice Do's

#### Why?

Make it easier to see the code behind our examples and learn what to do for best practices

Header Before:
Screen Shot 2022-07-05 at 2 44 12 PM

Header After:
Screen Shot 2022-07-05 at 2 42 05 PM

Best Practices Before:
Screen Shot 2022-07-05 at 2 47 22 PM

Best Practices After:
Screen Shot 2022-07-05 at 2 47 15 PM

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4579) and [Jira](https://jira.pinadmin.com/browse/GESTALT-4580)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [x] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/components/MainSectionCard.js [hunk 1]
```diff
@@ -88,10 +91,17 @@ function MainSectionCard({
       }}
     >
       {(title || type !== 'info') && (
-        <Box paddingY={1}>
+        <Box paddingY={1} display="flex" justifyContent="between">
           <Text weight="bold" color={TYPE_TO_COLOR[type]}>
             {cardTitle || capitalizeFirstLetter(type)}
           </Text>
+          {type === 'do' && code && (
```
Comment (ayeshakmaz) [inline docs/components/MainSectionCard.js:98]: Main change here for the Do's in Best Practices

File: docs/components/PageHeader.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ type Props = {|
   description?: string,
   fileName?: string, // only use if name !== file name
   folderName?: string, // only use if name !== file name and the link should point to a directory
+  showCode?: boolean,
```
Comment (ayeshakmaz) [inline docs/components/PageHeader.js:25]: add ability to toggle whether we show the code

File: docs/components/PageHeaderQualitySummary.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ export default function PageHeaderQualitySumary({ name }: Props): Node {
       <Flex wrap>
         {['figma', 'responsive', 'iOS', 'android', 'accessible'].map((item, index) => (
           <Flex key={`summary-${index}`}>
-            <Box paddingX={4}>
+            <Box paddingX={4} marginBottom={2}>
```
Comment (ayeshakmaz) [inline docs/components/PageHeaderQualitySummary.js:36]: Small spacing fix for small widths

**Review by AlbertCarreras — APPROVED** [review]
Thanks!

**Review by dangerismycat — APPROVED** [review]

**Pull request merged**